packages/@jsii/benchmarks/lib/index.ts

```diff
@@ -0,0 +1,44 @@
+import * as cp from 'child_process';
+import * as fs from 'fs-extra';
+import { Compiler } from 'jsii/lib/compiler';
+import { loadProjectInfo } from 'jsii/lib/project-info';
+import * as os from 'os';
+import * as path from 'path';
+
+import { Benchmark } from './benchmark';
+import { cdkv2_21_1 } from './constants';
+
+// Always run against the same version of CDK source
+const cdk = new Benchmark('Compile aws-cdk-lib@v2.21.1')
+  .setup(() => {
+    const sourceDir = fs.mkdtempSync(
+      path.join(os.tmpdir(), 'jsii-cdk-bench-snapshot'),
+    );
+    cp.execSync(`tar xf ${cdkv2_21_1}`, { cwd: sourceDir });
```

Comment:
Would prefer _not_ to shell out to `tar` here, this is kind of risky (`tar` on macOS versus `tar` on Linux versus the absence of `tar` on Windows...)

Reply:
So we ran out of memory in the gh action when calling `tar` within nodejs =(

Reply:
oh I'm dumb, I'll use a stream 😄